draft-ietf-httpbis-semantics-latest.xml

```diff
@@ -2782,88 +2776,62 @@ Sun Nov  6 08:49:37 1994         ; ANSI C's asctime() format
+   <xref target="RFC6125" x:fmt="of" x:sec="6"/> (for a reference identifier of
+   type URI-ID) unless the client has been specifically configured to accept
+   some other form of verification. For example, a client might be connecting
+   to a server whose address and hostname are dynamic, with an expectation that
+   the service will present a specific certificate (or a certificate matching
+   some externally defined reference identity) rather than one matching the
+   dynamic URI's origin server identifier.
 </t>
 <t>
-   In some cases, the URI is specified as an IP address rather than a
-   hostname. In this case, the iPAddress subjectAltName must be present
-   in the certificate and must exactly match the IP in the URI.
+   In special cases, it might be appropriate for
+   a client to simply ignore the server's identity, but it must be
+   understood that this leaves the connection open to active attack.
 </t>
 <t>
-   If the hostname does not match the identity in the certificate, user
-   oriented clients MUST either notify the user (clients MAY give the
-   user the opportunity to continue with the connection in any case) or
+   If the certificate is not valid for the URI's origin server,
+   a user agent &MUST; either notify the user (user agents &MAY; give the
+   user an option to continue with the connection in any case) or
    terminate the connection with a bad certificate error. Automated
    clients &MUST; log the error to an appropriate audit log (if available)
-   and SHOULD terminate the connection (with a bad certificate error).
-   Automated clients MAY provide a configuration setting that disables
+   and &SHOULD; terminate the connection (with a bad certificate error).
+   Automated clients &MAY; provide a configuration setting that disables
    this check, but &MUST; provide a setting which enables it.
 </t>
-<t>
-   Note that in many cases the URI itself comes from an untrusted
-   source. The above-described check provides no protection against
-   attacks where this source is compromised. For example, if the URI was
-   obtained by clicking on an HTML page which was itself obtained
-   without using HTTP/TLS, an on-path attacker could have replaced the
-   URI.  In order to prevent this form of attack, users should carefully
-   examine the certificate presented by the server to determine if it
-   meets their expectations.
-</t>
 </section>
 
 <section title="Identifying HTTPS Clients" anchor="https.identify.client">
 <t>
    Typically, the server has no external knowledge of what the client's
-   identity ought to be and so checks (other than that the client has a
+   identity ought to be; checks (other than that the client has a
    certificate chain rooted in an appropriate CA) are not possible. If a
    server has such knowledge (typically from some source external to
-   HTTP or TLS) it &SHOULD; check the identity as described above.
+   HTTP or TLS), it &SHOULD; check the identity as described above.
```

Comment:
"Above" is a little non-specific and mostly likely wrong (if it refers to using 6125). 

I think we can agree that anything more specific than "application-defined" is unlikely to match reality.

Reply:
I am going to remove this client certs section because it serves no purpose right now. It's more of an answer to an old FAQ, for which there isn't a standard.